src/modules/0.0.22/aws_security_group/index.ts

```diff
@@ -309,11 +308,6 @@ class SecurityGroupMapper extends MapperBase<SecurityGroup> {
           await this.module.securityGroup.db.update(e, ctx);
           // Make absolutely sure it shows up in the memo
           ctx.memo.db.SecurityGroup[e.groupId ?? ''] = e;
-          const rules = ctx?.memo?.cloud?.SecurityGroupRule ?? [];
```

Comment:
wouldn't this fix the issue?
```ts
const rules = Object.values(ctx?.memo?.cloud?.SecurityGroupRule ?? {});
```

Reply:
So that was the first change I tried (with `Object.values(rules).filter...` but it immediately broke the test suite...

Reply:
aren't we doing this in other places? should we create a task to investigate if this memo access is a problem in other modules?

Reply:
So a `git grep "memo?."` shows all of the places we do this, and everything else either accesses the memoized object in question by an ID, or calls `Object.values` like @aguillenv suggested I do here.

The reason why I am deleting simply deleting this code is that we got the desired behavior according to our test suite with this path literally throwing an error before, and it was only because of a race condition that the error was never surfaced.

When I did try to "fix" it it broke the test suite, instead, which is further proof that it should just be excised.

The question I now have is this:

Since this code never did anything, should I delete it from all prior supported versions and make the `@cloudId` change easier, or should I not touch prior versions and create a new `@cloudId` decorator that we switch to and eventually delete the old one?

Reply:
I'm onboard with fixing previous versions of the codebase and making the change easier